cpp/src/arrow/builder.cc

```diff
@@ -499,62 +541,63 @@ Status AdaptiveUIntBuilder::FinishInternal(std::shared_ptr<ArrayData>* out) {
   return Status::OK();
 }
 
-Status AdaptiveUIntBuilder::AppendValues(const uint64_t* values, int64_t length,
-                                         const uint8_t* valid_bytes) {
-  RETURN_NOT_OK(Reserve(length));
+Status AdaptiveUIntBuilder::AppendValuesInternal(const uint64_t* values, int64_t length,
+                                                 const uint8_t* valid_bytes) {
+  // See AdaptiveIntBuilder::AppendValuesInternal
+  constexpr int64_t max_chunk_size = 8192;
```

Comment:
Is this something that we should define somewhere as a global constant?

Reply:
Yes, will do.